### DISCUSSION TITLE
How use both watch and sync command with Docker Compose?

### DISCUSSION THREAD

**Discussion by alexkahler** [Taxel/PlexTraktSync#1623] [Support] [answered]
So far I have successfully added the sync and everything works between Plex and Trakt. However, I don't know how to proceed from here in regards to getting the watch command to work? Should I add it to the same container as the sync?

My initial yaml script is like this:

```
version: '2'
services:
  plextraktsync:
    image: ghcr.io/taxel/plextraktsync:latest
    command: sync
    container_name: plextraktsync
    restart: on-failure:2
    stdin_open: true
    tty: true
    volumes:
      - /share/appdata/plextraktsync:/app/config:rw
    environment:
      - PUID=1008
      - PGID=100
      - TZ=Europe/Copenhagen
```
I'm using a QNAP NAS with Container Station.

Note: I'm completely new to docker, so please ELI5 the shit out of this.

Thanks 🙏

**Comment by alexkahler** [comment]
I found a solution here: https://github.com/Taxel/PlexTraktSync/discussions/963#discussioncomment-3463194

So my compose script ended looking up like this:
```
version: '2'
services:
  plextraktsync:
    image: ghcr.io/taxel/plextraktsync:latest
    command: sync
    container_name: plextraktsync
    restart: on-failure:2
    stdin_open: true
    tty: true
    volumes:
      - /share/appdata/plextraktsync:/app/config:rw
    environment:
      - PUID=1008
      - PGID=100
      - TZ=Europe/Copenhagen
  plextraktsyncwatch:
    image: ghcr.io/taxel/plextraktsync:latest
    command: watch
    container_name: plextraktsyncwatch
    restart: unless-stopped
    stdin_open: true
    tty: true
    volumes:
      - /share/appdata/plextraktsync:/app/config:rw
    environment: #This is for the docker user, as described in https://www.linuxserver.io/blog/2017-09-17-how-to-setup-containers-on-qnap
      - PUID=1008
      - PGID=100
      - TZ=Europe/Copenhagen
```

I changed config.yml to this:
```
#settings for 'watch' command
watch:
  add_collection: true # Set to "true" because the sync only happens once every 3 hours.
  remove_collection: false
  #what video watched percentage (0 to 100) triggers the watched status
  scrobble_threshold: 80
  #true to scrobble only what's watched by you, false for all your PMS users
  username_filter: true
  #Show the progress bar of played media in terminal
  media_progressbar: true
```

As you might also note, QNAP container station YML validator is messed up, so it can't handle the Ofelia daemon scheduler. To circumvent this I added `0 */3 * * * docker start plextraktsync` to the [crontab file](https://wiki.qnap.com/wiki/Add_items_to_crontab). This will enable full sync for the library.

Please comment if you have any suggestions to improvements.

**Comment by scoiattolo49** [comment, marked as answer]
There is a maintained docker image available that does scrobble and cron based syncs all in one:
https://github.com/linuxserver-labs/docker-plextraktsync

This one (once a container is created) needs a single command run to go through the wizard and set up the authentication and then it does both watch/scrobble and cron based syncs. By default every 2 hours, but can be easily adjusted by editing a file in the config folder

**Reply by alexkahler** [reply]
Hi, just changed out the image. Works like advertised.
